### PR TITLE
fix(expansion): enforce stricter types for inputs

### PR DESCRIPTION
**Pull request by rafaelss95**

### CODE REVIEW

File: src/cdk/accordion/accordion-item.ts [hunk 1]
```diff
@@ -90,9 +90,9 @@ export class CdkAccordionItem implements OnDestroy {
 
   /** Whether the AccordionItem is disabled. */
   @Input()
-  get disabled() { return this._disabled; }
-  set disabled(disabled: any) { this._disabled = coerceBooleanProperty(disabled); }
-  private _disabled: boolean = false;
+  get disabled(): boolean { return this._disabled; }
+  set disabled(disabled: boolean) { this._disabled = coerceBooleanProperty(disabled); }
```
Comment (crisbeto) [inline src/cdk/accordion/accordion-item.ts:94]: The whole point of using coercion here is that you can pass in anything. The problem is that TS doesn't support different types for the getter and the setter.

Comment (rafaelss95) [reply src/cdk/accordion/accordion-item.ts:94]: Yep, I know. And I'm aware this is a breaking change and I wouldn't expect it to be merged until you reach v11 stages.

Comment (crisbeto) [reply src/cdk/accordion/accordion-item.ts:94]: If we were actually requiring these to be boolean, the `coerceBooleanProperty` call becomes redundant. I think that changing this defeats the purpose and we'd have to do it across the entire codebase.

Comment (rafaelss95) [reply src/cdk/accordion/accordion-item.ts:94]: You do actually use boolean for all boolean accessors, these here modified in this PR are outside the rule. It helps not only documentation, but the dev experience when accessing a _getter_, for example, you'd want a `boolean`, not `any`.

See these two examples for reference: https://github.com/angular/components/blob/cb8de6195eee87d7238fd6d3494b260b68ed5bd8/src/material/expansion/expansion-panel.ts#L107-L113
https://github.com/angular/components/blob/cfe37d18a67beddafd69ef95c2c605975e9b468f/src/cdk/drag-drop/directives/drop-list-group.ts#L36-L41

**Comment by jelbourn** [comment]
Caretaker note: Don't merge until v11 release train

**Comment by jelbourn** [comment]
@rafaelss95 could you add a "BREAKING CHANGE" note to the commit message?

**Comment by rafaelss95** [comment]
> @rafaelss95 could you add a "BREAKING CHANGE" note to the commit message?

Let me know if it's fine now.

**Comment by jelbourn** [comment]
@rafaelss95 just checked- the commit has to be a `fix` to show up in the changelog. You can also search (with `/`) through `git log` to see examples of other breaking changes blocks

**Comment by rafaelss95** [comment]
@annieyw is there any chance to merge this before v12? :)

**Pull request merged**